[BUG] test_hash_repartition_long_overflow_ansi_exception failed on 341DB

**Describe the bug**
A pipeline reported integration test `test_hash_repartition_long_overflow_ansi_exception` failed on 341DB.

Log:
```

[2024-10-15T11:00:30.712Z] =================================== FAILURES ===================================

[2024-10-15T11:00:30.712Z] ____________ test_hash_repartition_long_overflow_ansi_exception[17] ____________

[2024-10-15T11:00:30.712Z] [gw0] linux -- Python 3.10.12 /usr/bin/python

[2024-10-15T11:00:30.712Z] 

[2024-10-15T11:00:30.712Z] num_parts = 17

[2024-10-15T11:00:30.712Z] 

[2024-10-15T11:00:30.712Z]     @ignore_order(local=True)  # To avoid extra data shuffle by 'sort on Spark' for this repartition test.

[2024-10-15T11:00:30.712Z]     @pytest.mark.parametrize('num_parts', [17], ids=idfn)

[2024-10-15T11:00:30.712Z]     @allow_non_gpu(*non_utc_allow)

[2024-10-15T11:00:30.712Z]     def test_hash_repartition_long_overflow_ansi_exception(num_parts):

[2024-10-15T11:00:30.712Z]         data_gen = [('a', long_gen)]

[2024-10-15T11:00:30.712Z]         part_on = [f.col('a') + 15]

[2024-10-15T11:00:30.712Z]         conf = ansi_enabled_conf

[2024-10-15T11:00:30.712Z]     

[2024-10-15T11:00:30.712Z]         def test_function(spark):

[2024-10-15T11:00:30.712Z]             return gen_df(spark, data_gen, length=1024) \

[2024-10-15T11:00:30.712Z]                 .withColumn('plus15', f.col('a') + 15) \

[2024-10-15T11:00:30.712Z]                 .repartition(num_parts, f.col('plus15')) \

[2024-10-15T11:00:30.712Z]                 .withColumn('id', f.spark_partition_id()) \

[2024-10-15T11:00:30.712Z]                 .withColumn('hashed', f.hash(*part_on)) \

[2024-10-15T11:00:30.712Z]                 .selectExpr('*', 'pmod(hashed, {})'.format(num_parts))

[2024-10-15T11:00:30.712Z]     

[2024-10-15T11:00:30.712Z] >       assert_gpu_and_cpu_error(

[2024-10-15T11:00:30.712Z]             lambda spark: test_function(spark).collect(),

[2024-10-15T11:00:30.712Z]             conf=conf, error_message="ArithmeticException")

[2024-10-15T11:00:30.712Z] 

[2024-10-15T11:00:30.712Z] ../../src/main/python/repart_test.py:328: 

[2024-10-15T11:00:30.712Z] _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 

[2024-10-15T11:00:30.712Z] ../../src/main/python/asserts.py:664: in assert_gpu_and_cpu_error

[2024-10-15T11:00:30.712Z]     assert_spark_exception(lambda: with_cpu_session(df_fun, conf), error_message)

[2024-10-15T11:00:30.712Z] _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 

[2024-10-15T11:00:30.712Z] 

[2024-10-15T11:00:30.712Z] func = <function assert_gpu_and_cpu_error.<locals>.<lambda> at 0x7f03b0eb0c10>

[2024-10-15T11:00:30.712Z] error_message = 'ArithmeticException'

[2024-10-15T11:00:30.712Z] 

[2024-10-15T11:00:30.712Z]     def assert_spark_exception(func, error_message):

[2024-10-15T11:00:30.712Z]         """

[2024-10-15T11:00:30.712Z]         Assert that a specific Java exception is thrown

[2024-10-15T11:00:30.712Z]         :param func: a function to be verified

[2024-10-15T11:00:30.712Z]         :param error_message: a string such as the one produce by java.lang.Exception.toString

[2024-10-15T11:00:30.712Z]         :return: Assertion failure if no exception matching error_message has occurred.

[2024-10-15T11:00:30.712Z]         """

[2024-10-15T11:00:30.712Z] >       with pytest.raises(Exception) as excinfo:

[2024-10-15T11:00:30.712Z] E       Failed: DID NOT RAISE <class 'Exception'>

[2024-10-15T11:00:30.712Z] 

[2024-10-15T11:00:30.712Z] ../../src/main/python/asserts.py:650: Failed
```

--- a/integration_tests/src/main/python/repart_test.py
+++ b/integration_tests/src/main/python/repart_test.py
@@ -313,16 +313,16 @@ def test_hash_repartition_exact_longs_no_overflow(num_parts, is_ansi_mode):
 @pytest.mark.parametrize('num_parts', [17], ids=idfn)
 @allow_non_gpu(*non_utc_allow)
 def test_hash_repartition_long_overflow_ansi_exception(num_parts):
-    data_gen = [('a', long_gen)]
-    part_on = [f.col('a') + 15]
     conf = ansi_enabled_conf
 
     def test_function(spark):
-        return gen_df(spark, data_gen, length=1024) \
-            .withColumn('plus15', f.col('a') + 15) \
-            .repartition(num_parts, f.col('plus15')) \
+        df = gen_df(spark, [('a', long_gen)], length=1024)
+        maxVal = df.selectExpr("max(a) as m").head()['m']
+        overflowVal = (1 << 63) - maxVal
+        return df.withColumn('plus', f.col('a') + overflowVal) \
+            .repartition(num_parts, f.col('plus')) \
             .withColumn('id', f.spark_partition_id()) \
-            .withColumn('hashed', f.hash(*part_on)) \
+            .withColumn('hashed', f.hash(f.col('a') + overflowVal)) \
             .selectExpr('*', 'pmod(hashed, {})'.format(num_parts))
 
     assert_gpu_and_cpu_error(

